packages/components/src/higher-order/with-notices/index.js

```diff
@@ -25,48 +25,54 @@ export default createHigherOrderComponent( ( OriginalComponent ) => {
 	function Component( props, ref ) {
 		const [ noticeList, setNoticeList ] = useState( [] );
 
-		const noticeOperations = {
-			/**
-			 * Function passed down as a prop that adds a new notice.
-			 *
-			 * @param {Object} notice  Notice to add.
-			 */
-			createNotice: ( notice ) => {
-				const noticeToAdd = notice.id
-					? notice
-					: { ...notice, id: uuid() };
-				setNoticeList( ( current ) => [ ...current, noticeToAdd ] );
-			},
+		/**
+		 * Function passed down as a prop that adds a new notice.
+		 *
+		 * @param {Object} notice  Notice to add.
+		 */
+		const createNotice = useCallback( ( notice ) => {
+			const noticeToAdd = notice.id ? notice : { ...notice, id: uuid() };
+			setNoticeList( ( current ) => [ ...current, noticeToAdd ] );
+		}, [] );
 
-			/**
-			 * Function passed as a prop that adds a new error notice.
-			 *
-			 * @param {string} msg  Error message of the notice.
-			 */
-			createErrorNotice: ( msg ) => {
-				noticeOperations.createNotice( {
+		/**
+		 * Function passed as a prop that adds a new error notice.
+		 *
+		 * @param {string} msg  Error message of the notice.
+		 */
+		const createErrorNotice = useCallback(
+			( msg ) => {
+				createNotice( {
 					status: 'error',
 					content: msg,
 				} );
 			},
+			[ createNotice ]
```

Comment:
It seems like this might as well be `[]` too but not like it's going to harm anything as is.

Reply:
Yes, I did think of doing that, but as it technically relies on it, it didn't feel very clear as to why it's left out. If `createNotice` had other dependencies in future, would it be obvious enough that this would be affected too?

Reply:
Maybe that'd indicate it could be neater to memoize the whole `noticeOperations` object so there'd be one shared dependency list. While reviewing this I had experimented with doing that just for curiosity's sake but it doesn't seem of substantial benefit.

Reply:
Yes, I'm not certain why I abandoned that approach. It's probably reasonable to assume that if one operation changes then they all will, but maybe that's not the case?